src/examples/src/App.tsx

```diff
@@ -0,0 +1,96 @@
+import { create, tsx } from '@dojo/framework/core/vdom';
+import block from '@dojo/framework/core/middleware/block';
+import Outlet from '@dojo/framework/routing/Outlet';
+
+import readme from './readme.block';
+import getWidgetProperties, { PropertyInterface } from './properties.block';
+import getTheme from './theme.block';
+import code from './code.block';
+import configs, { getWidgetFileNames } from './config';
+import Menu from './Menu';
+import SideMenu from './SideMenu';
+import Example from './Example';
+import ThemeTable from './ThemeTable';
+import PropertyTable from './PropertyTable';
+
+import * as css from './App.m.css';
+
+const widgetFilenames = getWidgetFileNames(configs);
+
+interface AppProperties {
+	includeDocs: boolean;
+}
+
+const factory = create({ block }).properties<AppProperties>();
+
+interface Content {
+	[index: string]: string;
+}
+
+export default factory(function App({ properties, middleware: { block } }) {
+	const { includeDocs } = properties();
+	const widgets = Object.keys(configs).sort();
+	let widgetReadmeContent: Content = {};
+	let widgetExampleContent: Content = {};
+	let widgetProperties: { [index: string]: PropertyInterface[] } = {};
+	let widgetThemeClasses: { [index: string]: string[] } = {};
+	if (includeDocs) {
+		widgetReadmeContent = block(readme)() || {};
+		widgetExampleContent = block(code)() || {};
+		widgetProperties = block(getWidgetProperties)(widgetFilenames) || {};
+		widgetThemeClasses = block(getTheme)(widgetFilenames) || {};
+	}
+	return (
+		<div classes={[css.root]}>
+			<Menu widgetNames={widgets} />
+			<main classes={[css.main]}>
+				<Outlet
+					id="landing"
+					renderer={() => {
+						return <div>Widget Examples!</div>;
+					}}
+				/>
+				<Outlet
+					id="example"
+					renderer={({ params }) => {
+						const { widget: widgetName, example: exampleName } = params;
+						const widgetConfig = configs[widgetName];
+						const { overview, examples = [] } = widgetConfig;
+						const isBasic = exampleName === 'basic';
+						const readmeContent = widgetReadmeContent[widgetName];
+						const example = isBasic
+							? overview.example
+							: examples.find(
+									(example) => example.filename.toLowerCase() === exampleName
+							  );
+						if (!example) {
+							return null;
+						}
+						const widgetPath = `${widgetName}/${example.filename}`;
+						const content =
+							widgetExampleContent[`${widgetPath}.tsx`] ||
+							widgetExampleContent[`${widgetPath}.ts`];
+						const propertyInterface = widgetProperties[widgetName];
+						const themeClasses = widgetThemeClasses[widgetName];
+						return (
+							<virtual key={widgetPath}>
```

Comment:
What's `<virtual>` doing here? Never seen this before.

Reply:
`<virtual>` is a special node that can be used when you want to return an array in tsx, there is more information here https://dojo.io/learn/creating-widgets/rendering-widgets#tsx-widget-example